portal/patient/scripts/app/onsitedocuments.js

```diff
@@ -656,12 +656,20 @@ var page = {
                             if (page.isFrameForm) {
                                 // a layout form
                                 if (page.lbfFormName) {
-                                    // iframe from template directive {EncounterDocument:LBFxxxxx}
-                                    let url = webRoot + "/interface/forms/LBF/new.php" + "" +
-                                        "?isPortal=" + encodeURIComponent(isPortal ? 1 : 0) +
-                                        "&formOrigin=" + encodeURIComponent(page.formOrigin) +
-                                        "&formname=" + encodeURIComponent(page.lbfFormName) + "&id=0";
-
+                                    let url = '';
+                                    if (page.lbfFormName.startsWith('LBF')) {
+                                        // iframe from template directive {EncounterDocument:LBFxxxxx} for a LBF form
+                                        url = webRoot + "/interface/forms/LBF/new.php" + "" +
+                                            "?isPortal=" + encodeURIComponent(isPortal ? 1 : 0) +
+                                            "&formOrigin=" + encodeURIComponent(page.formOrigin) +
+                                            "&formname=" + encodeURIComponent(page.lbfFormName) + "&id=0";
+                                    } else {
+                                        // iframe from template directive {EncounterDocument:xxxxx} for a native form
+                                        url = webRoot + "/interface/forms/" + encodeURIComponent(page.lbfFormName) + "/new.php" +
+                                            "?isPortal=" + encodeURIComponent(isPortal ? 1 : 0) +
+                                            "&formOrigin=" + encodeURIComponent(page.formOrigin) +
+                                            "&formname=" + encodeURIComponent(page.lbfFormName) + "&id=0";
+                                    }
```

Comment:
This is exactly the approach I'd taken. Besides the audit, we may want to send a notification to user for form completion.
This is something I want to do a better job at doing meaning, notifications in portal. Not only user but SMS/email to patient via event.

Looking good so far and let me know if I can help.